### ISSUE TITLE
[Feature Request] Need Matmul Attention layer instead of Einsum to support GPU running

### ISSUE THREAD

**Issue by MoFHeka** [NVIDIA/JAX-Toolbox#502]
Einsum kernel in Praxis couldn't' be lowered to cudnn GEMM. The computing performance is seriously affected. Jax version Attention layer much slower than Tensorflow version.

**Comment by nouiz** [comment]
Why do you say cudnn GEMM isn't used? Normally it should. Can you provide an example where cudnn gemm isn't used in such a case?

**Comment by MoFHeka** [comment]
@nouiz Yes, TransformerEngine did use cudnn GEMM. But JAX(Flax or Praxis) attention layers was constructed by Einsum kernels, which couldn't' be lowered to cudnn GEMM and the latest cudnn XLA FMHA kernel. When running attention layers in GPU, it could be only transformed to triton kernel according the XLA dump log...

TE currently only supports a limited number of transformer models (such as MOE is difficult to support) and does not yet support LORA SFT. So it may be necessary to optimize the layer composition of the Jax ecosystem.

Sorry, I'm not sure where to put the requirement because it doesn't look like the TE team should be responsible for it. As I understand it, the TE team is only responsible for the 'custom call' part of jax.

![BossHi-20240128172529](https://github.com/NVIDIA/JAX-Toolbox/assets/90189118/497f4c45-a8e2-44db-98b9-f6c1ad3be8a9)

**Comment by nouiz** [comment]
I think you will be interested by this PR: https://github.com/google/jax/pull/18814

**Comment by MoFHeka** [comment]
Cool, thank you! 
It would be nice if someone could also change the code in jax wrapper components like praxis, flax, etc., since they are now written in einsum.
